File: lib/enums/meal_time_type.dart
import 'package:hive/hive.dart'; // Added

part 'meal_time_type.g.dart'; // Added

/// Enum representing different meal times of the day
@HiveType(typeId: 1) // Added
enum MealTimeType {
  @HiveField(0) // Added
  breakfast,
  @HiveField(1) // Added
  lunch,
  @HiveField(2) // Added
  dinner,
  @HiveField(3) // Added
  snack;

  /// Returns a human-readable name for the meal type
  String get displayName {
    switch (this) {
      case MealTimeType.breakfast:
        return '朝食';
      case MealTimeType.lunch:
        return '昼食';
      case MealTimeType.dinner:
        return '夕食';
      case MealTimeType.snack:
        return '間食';
    }
  }

  /// Converts a string value to the corresponding MealTimeType
  static MealTimeType fromString(String value) {
    return MealTimeType.values.firstWhere(
      (type) => type.name.toLowerCase() == value.toLowerCase(),
      orElse: () => MealTimeType.snack,
    );
  }
}

File: Tonton/Tonton/Models/MealTimeType.swift
//
//  MealTimeType.swift
//  TonTon
//
//  Enum representing different meal times of the day
//  Migrated from Flutter MealTimeType enum
//

import Foundation

enum MealTimeType: String, CaseIterable, Codable {
    case breakfast = "breakfast"
    case lunch = "lunch"
    case dinner = "dinner"
    case snack = "snack"
    
    /// Returns a human-readable name for the meal type
    var displayName: String {
        switch self {
        case .breakfast:
            return "Breakfast"
        case .lunch:
            return "Lunch"
        case .dinner:
            return "Dinner"
        case .snack:
            return "Snack"
        }
    }
    
    /// Returns localized display name (for Japanese/English support)
    func localizedDisplayName(for locale: Locale) -> String {
        if locale.identifier.hasPrefix("ja") {
            switch self {
            case .breakfast: return "朝食"
            case .lunch: return "昼食"
            case .dinner: return "夕食"
            case .snack: return "間食"
            }
        }
        return displayName
    }
    
    /// Converts a string value to the corresponding MealTimeType
    static func fromString(_ value: String) -> MealTimeType {
        return MealTimeType(rawValue: value.lowercased()) ?? .snack
    }
}